addons/stock/models/stock_quant.py

```diff
@@ -646,6 +646,36 @@ def _compute_complete_name(self):
             res[package.id] = name
         return res
 
+    def _search_location(self, operator, value):
+        if value:
+            packs = self.search([('quant_ids.location_id', operator, value)])
+        else:
+            packs = self.search([('quant_ids', operator, value)])
```

Comment:
Location is required on the quants anyways.  It is just important to be able to find packages without quants as they can be used instead of creating a new package.  (so you scan them, ...)